File: stack.java
class stack
{
	public static void main(String args[]) {
		int size = 16;
		while(size < 1000000000) {
			do_stack(size);
			size *= 2;
		}
	}
	static void do_stack(int size) {
		int a[];
		a = new int[size];
		int sum = 0;
		long start = System.nanoTime();
		for (int i = 0; i < size; i++) {
			a[i] = 1;
		}
		for (int i = size - 1; i >= 0; i--) {
			sum += a[i];
		}
		long end = System.nanoTime();
		System.out.println((end - start) / 1000000000f);
	}
}

File: 課題1/stack.rb
def stack(n)
	a = Array.new(2 ** n, 1)
	sum = 0
	t0 = Time.now
	for i in 0 .. 2 ** n - 1
		a[i] = 1
	end
	for i in 2 ** n - 1 .. 0
		sum += a[i]
	end
	t1 = Time.now

	puts "#{(t1 - t0)}"
end

def main(n)
	i = 4
	while 2 ** i < 10000000
		stack(i)
		i += 1
	end
end 

